docs/Guide/preconditions/creating-your-own-preconditions.mdx

```diff
@@ -1,6 +1,225 @@
+  }
+}
+```
+
+</TabItem>
+</Tabs>
+
+## Using Preconditions in Commands
+
+To attach a precondition to a command, you simply have to input its name in an array in the command's
+[`preconditions`][preconditions-option] option.
+
+<Tabs groupId="language-choice">
+<TabItem value="javascript" label="JavaScript" default>
+
+```javascript {7}
+const { Command } = require('@sapphire/framework');
+
+module.exports = class PingCommand extends Command {
+  constructor(context) {
+    super(context, {
+      // ...
+      preconditions: ['OwnerOnly']
+    });
+  }
+};
+```
+
+</TabItem>
+<TabItem value="ESM" label="ESM">
+
+```javascript {7}
+import { Command } from '@sapphire/framework';
+
+export class PingCommand extends Command {
+  constructor(context) {
+    super(context, {
+      // ...
+      preconditions: ['OwnerOnly']
+    });
+  }
+}
+```
+
+</TabItem>
+<TabItem value="typescript" label="TypeScript">
+
+```typescript {7}
+import { Command } from '@sapphire/framework';
+
+export class PingCommand extends Command {
+  public constructor(context: Command.Context) {
+    super(context, {
+      // ...
+      preconditions: ['OwnerOnly']
+    });
+  }
+}
+```
+
+</TabItem>
+</Tabs>
+
+Now, if someone who is not the bot owner executes the ping command, nothing will happen!
+
+:::caution
+
+For TypeScript users, there's an extra step to make this work. To increase the security of Sapphire's types, you'll need
+to augment Sapphire's [`Preconditions`][preconditions-interface] interface. Please see an official example
+[here][preconditions-augment].
+
+<!--
+  TODO: Once we have a dedicated TypeScript page this should link to that instead of a code example.
+  Including why augmenting with `never` works and the difference between that and augmenting with an object.
+-->
+
+:::
+
+By default, no error message will be sent or logged when a command is denied because of a precondition. To learn how to
+configure this, please read [Reporting Precondition Failures][reporting-precondition-failure].
+
+## Advanced Usage
+
+Sapphire also has a builtin system for advanced conditional precondition logic through nested arrays. By default, all
+preconditions in the given array must pass for the command to be run. However, you can use nested arrays to create `OR`
+functionality. This could be useful if you'd like a command to be run if the user is either a moderator _or_ an admin.
+
+Furthermore, if you create a nested array within a nested array, you'll receive `AND` functionality once more. Arrays
+can be nested infinitely with the same pattern for optimal control over your preconditions.
+
+Consider the following array of preconditions:
+
+:::warning
+
+None of the above preconditions are builtin, and you'd have to create them separately.
+
+:::
+
+```js
+[['AdminOnly', ['ModOnly', 'OwnerOnly']], 'InVoiceChannel'];
+```
+
+For a command with these preconditions to pass the denial checks, the `InVoiceChannel` precondition must pass, as well
```

Comment:
TODO for us Devs: Maybe we can borrow a page from prisma's $and/$or/$not and provide that as an option too because by the gods this array stuff is barely readable @sapphiredev/developers

Reply:
I've never liked it either but in the infinite wisdom of @kyranet he thinks it's readable.

Reply:
Well TOOOO BAD he's not the only one making decisions :D
Can you make an issue in framework for this?

Reply:
https://github.com/sapphiredev/framework/issues/334